### ISSUE TITLE
newobject changes semantics of bool()

### ISSUE THREAD

**Issue by tsznuk** [PythonCharmers/python-future#211]
Consider the following code:

```
from builtins import object

class ResultSet(object):
    def __len__(self):
        return 0

print(bool(ResultSet()))
```

On Python3, this prints `False`. On Python2 - `True`.

The reason is that future.types.newobject redefines `__nonzero__` to behave as follows:
- If `__bool__` is available, it is called
- Otherwise `True` is returned - presence of `__len__` is not considered.

**Comment by edschofield** [comment]
Thanks, Tadeusz! I confirm this bug in v0.15.2.

**Comment by edschofield** [comment]
@tsznuk: I have uploaded a fix to the `v0.15.x` branch. Could you please verify that `newobject` now preserves all the right semantics of `bool()`?

**Comment by tsznuk** [comment]
As far as I can tell the code now correctly implements truth testing as described in Python docs.

Thanks for fixing this!
